emailCrawler.java

```diff
@@ -31,14 +24,13 @@
 
 	public static void main(String args[]) throws IOException {
 		String url = "http://" + args[0];
-		// this list is used to store all of urls which can be discovered in
-		// home page. Actually, I can write this program
-		// by using DFS so that I can find more discoverable pages on discovered
-		// pages. But I don't do it for that it may go into a infinite loop.
+		// this list is used to store all of urls which can be discovered in home page. Actually, I can write this 
+		// program by using DFS so that I can find more discoverable pages on pages which we have discovered. 
+		// But here, I don't do it for that it may go into a infinite loop.
```

Comment:
Yes, at the beginning, I have though about the hard limit but I think it's not a flexible way and it's not good for future maintenance.